WordPress/src/main/java/org/wordpress/android/ui/activitylog/ActivityLogDetailFragment.kt

```diff
@@ -63,22 +66,18 @@ class ActivityLogDetailFragment : Fragment() {
             activityCreatedDate.text = activityLogModel?.createdDate
             activityCreatedTime.text = activityLogModel?.createdTime
 
-            val rewindAndFinish = activityLogModel?.rewindAction?.let {
-                rewindAction -> {
-                    if (rewindAction()) activity?.finish()
-                }
-            }
-            activityRewindButton.setOnClickListener(rewindAndFinish)
+            activityRewindButton.setOnClickListener(activityLogModel?.rewindAction)
         })
+
         viewModel.rewindAvailable.observe(this, Observer { available ->
             activityRewindButton.visibility = if (available == true) View.VISIBLE else View.GONE
         })
-        viewModel.start(site, activityLogId)
-    }
 
-    override fun onDestroy() {
-        viewModel.stop()
-        super.onDestroy()
+        viewModel.showRewindDialog.observe(this, Observer<ActivityLogDetailModel> {
+            onRewindButtonClicked(it!!)
```

Comment:
I wouldn't use `!!`, I think it's better to either do `it?.let {onRewindButtonClicked(it)` or crash with a meaningful message (if you intended to crash)